File: docs/guides/java/src/test/java/com/algolia/searchWithOptionalFilters.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;
import java.util.List;

public class searchWithOptionalFilters {

  private static final List<String> labels = List.of(
    /* Your labels */
  );

  private static OptionalFilters reduceLabelsToFilters(List<String> labels) {
    return OptionalFilters.of(""); // Implement your logic here
  }

  public static void main(String[] args) throws Exception {
    try (SearchClient client = new SearchClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")) {
      OptionalFilters optionalFilters = reduceLabelsToFilters(labels);
      SearchParams searchParams = new SearchParamsObject().setQuery("<YOUR_SEARCH_QUERY>").setOptionalFilters(optionalFilters);

      client.searchSingleIndex("<YOUR_INDEX_NAME>", searchParams, Hit.class);
    } catch (Exception e) {
      System.out.println("An error occurred: " + e.getMessage());
    }
  }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/searchWithOptionalFilters.kt
import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*
import com.algolia.client.transport.*

val labels: List<String> = listOf() // A list of labels

val reduceLabelsToFilters: (List<String>) -> OptionalFilters = {
  OptionalFilters.of("") // Implement your logic here
}

suspend fun searchWithOptionalFilters() {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  val optionalFilters = reduceLabelsToFilters(labels)
  val searchParams =
    SearchParamsObject(query = "<YOUR_SEARCH_QUERY>", optionalFilters = optionalFilters)

  client.searchSingleIndex(indexName = "<YOUR_INDEX_NAME>", searchParams = searchParams)
}
